src/app/main/component/eco-news/components/news-list/news-list.component.ts

```diff
@@ -103,25 +119,127 @@ export class NewsListComponent implements OnInit, OnDestroy {
     if (this.tagsList !== value) {
       this.tagsList = value;
     }
-    this.hasNext = true;
-    this.currentPage = 0;
     this.dispatchStore(true);
   }
 
+  cancelSearch(): void {
+    if (this.searchNewsControl.value.trim() === '') {
+      this.searchToggle = false;
+    } else {
+      this.searchNewsControl.setValue('');
+    }
+  }
+
+  search(): void {
```

Comment:
A more appropriate name would reflect this toggling behavior and a better name would be something like with prefix `is` or `has` for `boolean` type:
```
toggleSearch(): void {
    this.isSearchVisible = !this.isSearchVisible;
}
```